### PR TITLE
LF-3637 (2): Update the date inputs used in date range selector [DateRange class]

### PR DESCRIPTION
**Pull request by SayakaOno**
**Description**

- install `vitest`
- add `test` script to run tests
- Create `DateRange` class that calculates `startDate` and `endDate` for each date range option. The class handles both week start day - Monday and Sunday, and tests were written for both.

Jira link: https://lite-farm.atlassian.net/browse/LF-3637

**Type of change**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

```
pnpm test src/tests/dateRange.test.js
```

- [x] Passes test case
- [ ] UI components visually reviewed on desktop view
- [ ] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [x] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

**Review by antsgar — APPROVED** [review]
Run the tests and it's working great, nice work! 🎉

File: packages/webapp/src/util/date.jsx [hunk 1]
```diff
@@ -46,3 +46,14 @@ export const getLocalDateInYYYYDDMM = (date = new Date()) => {
...
 };
+
+/**
+ * Returns a date that the given days is added to the given date.
+ * @param {date} Date ex. new Date(2023, 10, 1)
+ * @param {number} days days to add. Could be a negative number.
+ * @returns {string} Date
+ */
+export function addDaysToDate(date, days) {
+  const newDate = new Date(date.getTime());
```
[the remark this answers is not in the document]
Comment (SayakaOno) [reply packages/webapp/src/util/date.jsx:58]: Thank you for reviewing!
I thought that the given date would be unintentionally modified by `setDate()`, so I "copied" it first. I hope what I did is correct...

[the remark this answers is not in the document]
Comment (kathyavini) [reply packages/webapp/src/util/date.jsx:58]: Ohhh! I never think about Dates being passed by reference, but of course they are objects!

File: packages/webapp/src/tests/dateRange.test.js [hunk 2]
```diff
@@ -0,0 +1,216 @@
+/*
```
[the remark this answers is not in the document]
Comment (kathyavini) [reply packages/webapp/src/tests/dateRange.test.js:1]: I know, they are sooooo readable!!!! 😍

File: packages/webapp/src/util/date.jsx [hunk 3]
```diff
@@ -46,3 +46,15 @@ export const getLocalDateInYYYYDDMM = (date = new Date()) => {
   const [year, month, day] = [date.getFullYear(), date.getMonth() + 1, date.getDate()];
   return getDateInYYYYMMDD(year, month, day);
 };
+
```
Comment (kathyavini) [inline packages/webapp/src/util/date.jsx:49]: Is the `.jsx` extension here a holdover from a previous version of the file?

**Review by kathyavini — APPROVED** [review]
It looks great!! Lovely clean tests and functions 👌

File: packages/webapp/src/util/date.jsx (hunk 3 above)
Comment (SayakaOno) [reply packages/webapp/src/util/date.jsx:49]: Thank you for reviewing! I probably created the file by copying a component file. `useUnit`hook  still has `.jsx` too 😅 I will fix them in my next PR!

**Pull request merged**